### DISCUSSION TITLE
ERC Registry discussion

### DISCUSSION THREAD

**Discussion by david-bakin-sl** [hashgraph/hedera-smart-contracts#1035] [Ideas]
Discussing #1028 - [EPIC] Phase 1: ERC Registry

**Comment by david-bakin-sl** [comment]
Phase 1 could (should?) include an "incremental" mode not mentioned in current design doc:  Take a starting date or starting low entity number plus the existing .json files and the tool will scan from there and append (merge) new results into the json files.

**Comment by quiet-node** [comment]
Thank you, @david-bakin-sl. You're absolutely correct—the documentation does not explicitly state this, but it describes how the `/contracts&order=asc` endpoint operates. The endpoint retrieves a batch of contracts (up to a maximum of 100 contracts) in ascending order, from oldest to newest. Within the response from the mirror node, a `next` pointer is provided for pagination, allowing retrieval of subsequent batches. This effectively implements the incremental mode you mentioned.

The mirror node endpoint uses contract entities for incremental retrieval, which can serve as an alternative to the starting "date" approach you proposed. 

Excellent observation—thank you for pointing that out!

**Reply by Nana-EC** [reply]
Agreed, but notably eventually we should also support back filling.
Example you run this for 2024 because your tool prioritizes newer tokens.
Later when you have time and resource you may want to start from genesis and fill in as you make progress.
There's a few ways to do it but we should consider it for Phase 2 or 3.

**Reply by quiet-node** [reply]
Ahh I see what you mean! Yeah probably we can have like a sorting mechanism and sort the records from oldest to latest and that might help.
